android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/views/DynamicallyOrientingNavigationView.kt

```diff
@@ -22,49 +34,68 @@ import com.stadiamaps.ferrostar.maplibreui.runtime.navigationMapViewCamera
  * @param modifier The modifier to apply to the view.
  * @param styleUrl The MapLibre style URL to use for the map.
  * @param camera The bi-directional camera state to use for the map.
+ * @param navigationCamera The default camera state to use for navigation. This is applied on launch
+ *   and when centering.
  * @param viewModel The navigation view model provided by Ferrostar Core.
  * @param locationRequestProperties The location request properties to use for the map's location
  *   engine.
+ * @param config The configuration for the navigation view.
  * @param content Any additional composable map symbol content to render.
- * @param orientation The orientation of the device. Defaults to the current device orientation, but
- *   can be overridden (e.g. with Configuration.ORIENTATION_LANDSCAPE or
- *   Configuration.ORIENTATION_PORTRAIT).
  */
 @Composable
 fun DynamicallyOrientingNavigationView(
     modifier: Modifier,
-    orientation: Int = LocalConfiguration.current.orientation,
     styleUrl: String,
     camera: MutableState<MapViewCamera> = rememberSaveableMapViewCamera(),
     navigationCamera: MapViewCamera = navigationMapViewCamera(),
     viewModel: NavigationViewModel,
     locationRequestProperties: LocationRequestProperties =
         LocationRequestProperties.NavigationDefault(),
+    config: VisualNavigationViewConfig = VisualNavigationViewConfig.Default(),
     onTapExit: (() -> Unit)? = null,
     content: @Composable @MapLibreComposable() ((State<NavigationUiState>) -> Unit)? = null,
 ) {
-  when (orientation) {
-    Configuration.ORIENTATION_LANDSCAPE -> {
-      LandscapeNavigationView(
-          modifier,
-          styleUrl,
-          camera,
-          navigationCamera,
-          viewModel,
-          locationRequestProperties,
-          onTapExit,
-          content)
-    }
-    else -> {
-      PortraitNavigationView(
-          modifier,
-          styleUrl,
-          camera,
-          navigationCamera,
-          viewModel,
-          locationRequestProperties,
-          onTapExit,
-          content)
+  SystemUIDisposableEffect()
+
+  val orientation = LocalConfiguration.current.orientation
+  val isLandscape = orientation == Configuration.ORIENTATION_LANDSCAPE
+
+  Box(modifier) {
+    NavigationMapView(
+        styleUrl,
+        mapControlsFor(isLandscape = isLandscape, isArrivalExpanded = onTapExit != null),
+        camera,
+        navigationCamera,
+        viewModel,
+        locationRequestProperties,
+        onMapReadyCallback = { camera.value = navigationCamera },
+        content)
+
+    when (orientation) {
+      Configuration.ORIENTATION_LANDSCAPE -> {
+        LandscapeNavigationOverlayView(
+            modifier =
+                Modifier.fillMaxSize()
+                    // TODO: DO NOT MERGE - Finalize w/ device
```

Comment:
Noted 😂 These sorts of things always risk breakage...

Reply:
Exactly why I put this there. I just tested on a device and think it looks great 👍